Mono.Debugging/Mono.Debugging.Client/StackFrame.cs

```diff
@@ -122,63 +128,110 @@ internal void Attach (DebuggerSession debugSession)
 		/// <summary>
 		/// Gets the full name of the stackframe. Which respects Session.EvaluationOptions.StackFrameFormat
 		/// </summary>
-		public virtual string FullStackframeText {
-			get {
-				var methodNameBuilder = new StringBuilder (this.SourceLocation.MethodName);
-				var options = session.EvaluationOptions.Clone ();
-				if (options.StackFrameFormat.ParameterValues) {
-					options.AllowMethodEvaluation = true;
-					options.AllowToStringCalls = true;
-					options.AllowTargetInvoke = true;
-				} else {
-					options.AllowMethodEvaluation = false;
-					options.AllowToStringCalls = false;
-					options.AllowTargetInvoke = false;
-				}
+		[Obsolete]
+		public string FullStackframeText {
+			get { return GetFullStackFrameText (); }
+		}
 
-				var args = this.GetParameters (options);
+		public string GetFullStackFrameText ()
+		{
+			return GetFullStackFrameText (session.EvaluationOptions);
+		}
 
-				//MethodName starting with "["... it's something like [ExternalCode]
-				if (!this.SourceLocation.MethodName.StartsWith ("[", StringComparison.Ordinal)) {
-					if (options.StackFrameFormat.Module && !string.IsNullOrEmpty (this.FullModuleName)) {
-						methodNameBuilder.Insert (0, System.IO.Path.GetFileName (this.FullModuleName) + "!");
-					}
-					if (options.StackFrameFormat.ParameterTypes || options.StackFrameFormat.ParameterNames || options.StackFrameFormat.ParameterValues) {
-						methodNameBuilder.Append ("(");
-						for (int n = 0; n < args.Length; n++) {
-							if (args [n].IsEvaluating) {
-								var manualReset = new ManualResetEvent (false);
-								EventHandler updated = (s, e) => {
-									manualReset.Set ();
-								};
-								args [n].ValueChanged += updated;
-								if (args [n].IsEvaluating)
-									if (!manualReset.WaitOne (2000))
-										session.OnDebuggerOutput (true, "Timeout evaluating parameters for StackFrame.");
-								args [n].ValueChanged -= updated;
-							}
-							if (n > 0)
-								methodNameBuilder.Append (", ");
-							if (options.StackFrameFormat.ParameterTypes) {
-								methodNameBuilder.Append (args [n].TypeName);
-								if (options.StackFrameFormat.ParameterNames)
-									methodNameBuilder.Append (" ");
-							}
-							if (options.StackFrameFormat.ParameterNames)
-								methodNameBuilder.Append (args [n].Name);
-							if (options.StackFrameFormat.ParameterValues) {
-								if (options.StackFrameFormat.ParameterTypes || options.StackFrameFormat.ParameterNames)
-									methodNameBuilder.Append (" = ");
-								var val = args [n].Value ?? "";
-								methodNameBuilder.Append (val.Replace ("\r\n", " ").Replace ("\n", " "));
+		public virtual string GetFullStackFrameText (EvaluationOptions options)
```

Comment:
```suggestion
		public string GetFullStackFrameText (EvaluationOptions options)
```

Reply:
Can't because the DotNetCore backend overrides this.